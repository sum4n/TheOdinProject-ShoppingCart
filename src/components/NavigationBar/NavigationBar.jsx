import styles from "./NavigationBar.module.css";
import { Link } from "react-router-dom";

const NavigationBar = ({ totalItemsInCart }) => {
  return (
    <nav className={styles.nav}>
      <p className={styles.title}>The shopping project</p>
      <ul className={styles.list}>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="shopPage">Products</Link>
        </li>
        <li>
          <a href="#">Contact</a>
        </li>
      </ul>
      <p>Cart: {totalItemsInCart}</p>
    </nav>
  );
};

export default NavigationBar;
