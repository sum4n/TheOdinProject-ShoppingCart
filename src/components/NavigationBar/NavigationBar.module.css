.nav {
  display: flex;
  justify-content: space-between;
  gap: 40px;
  border: 1px solid rgba(0, 183, 255, 0.164);
  padding-left: 2%;
  padding-right: 2%;
  margin: 0% 2% 0% 2%;
  background-color: rgb(252, 175, 61);
  box-shadow: 0px 0px 8px rgb(63, 63, 63);
  border-radius: 4px;
}

.title {
  flex-grow: 1;
}

a {
  color: blue;
  text-decoration: none;
}

a:hover {
  color: red;
}

.list {
  display: flex;
  gap: 14px;
  flex-grow: 0;
  /* border: 1px solid blue; */
}

li {
  list-style: none;
}
