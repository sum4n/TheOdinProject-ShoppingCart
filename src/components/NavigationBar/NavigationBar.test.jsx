import { render, screen } from "@testing-library/react";
import { describe, expect } from "vitest";
import NavigationBar from "./NavigationBar";
import { MemoryRouter } from "react-router-dom";

describe("NavigationBar component", () => {
  it("renders with  5 links", () => {
    render(
      <MemoryRouter>
        <NavigationBar totalItemsInCart={1} />
      </MemoryRouter>
    );

    expect(screen.getAllByRole("link")).toHaveLength(5);
    expect(
      screen.getByRole("link", { name: /the shopping project/i })
    ).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /home/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /products/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /contact/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /cart/i })).toBeInTheDocument();
  });

  it("displays 8 cart items", () => {
    render(<NavigationBar totalItemsInCart={8} />, { wrapper: MemoryRouter });

    expect(screen.getByRole("link", { name: /cart: 8/i })).toBeInTheDocument();
  });

  it("matches snapshot with 10 items in the cart", () => {
    const { container } = render(
      <MemoryRouter>
        <NavigationBar totalItemsInCart={10} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
