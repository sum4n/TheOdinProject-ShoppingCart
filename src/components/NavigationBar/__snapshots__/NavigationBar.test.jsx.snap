// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationBar component > renders with 10 cart items 1`] = `
<div>
  <nav
    class="_nav_982763"
  >
    <p
      class="_title_982763"
    >
      The shopping project
    </p>
    <ul
      class="_list_982763"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/shopPage"
        >
          Products
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Contact
        </a>
      </li>
    </ul>
    <p>
      Cart: 
      10
    </p>
  </nav>
</div>
`;
